client/shared/src/testing/integration/polly/CdpAdapter.ts

```diff
@@ -99,39 +104,69 @@ export class CdpAdapter extends PollyAdapter {
         // eslint-disable-next-line @typescript-eslint/ban-ts-comment
         // @ts-ignore
         super(polly)
-        this.page = this.options.page
+        this.browser = this.options.browser
     }
 
     /**
-     * Called when connecting to a Puppeteer page. Sets up CDP request
-     * interception using the CDP "Fetch domain".
+     * Called when the adapter is connected to. Sets up CDP request
+     * interception for all browser pages using the CDP "Fetch domain".
      */
     public async onConnect(): Promise<void> {
-        this.cdpSession = await this.page.target().createCDPSession()
+        // Create CDP sessions for all existing targets
+        const targets = await this.browser.targets()
+
+        await Promise.all(
+            targets.map(async target => {
+                await this.setupCDPSessionForTarget(target)
+            })
+        )
+
+        // Listen for future pages, create CDP session on creation
+        this.browser.on('targetcreated', target => {
+            this.setupCDPSessionForTarget(target).catch(noop)
+        })
+
+        // Delete associated CDP session when target is destroyed
+        this.browser.on('targetdestroyed', target => {
+            const cdpSession = this.cdpSessions.get(target)
+            this.cdpSessions.delete(target)
+            cdpSession?.detach().catch(noop)
+        })
+    }
+
+    /**
+     * Creates a CDP session for the given target and listens
+     * for paused requests
+     */
+    private async setupCDPSessionForTarget(target: Puppeteer.Target): Promise<void> {
+        const cdpSession = await target.createCDPSession()
+        this.cdpSessions.set(target, cdpSession)
 
         // TODO: This is where we narrow down the interception with patterns.
         // Request and respond stages are independent, so we can set a different
         // set of patterns for each.
         const fetchEnableRequest: Protocol.Fetch.EnableRequest = {
             patterns: [{ requestStage: 'Request' }, { requestStage: 'Response' }],
         }
-        await this.cdpSession.send('Fetch.enable', fetchEnableRequest)
+        await cdpSession.send('Fetch.enable', fetchEnableRequest)
 
-        this.cdpSession.on('Fetch.requestPaused', (event: Protocol.Fetch.RequestPausedEvent): void => {
+        cdpSession.on('Fetch.requestPaused', (event: Protocol.Fetch.RequestPausedEvent): void => {
             const isInResponseStage = eventIsInResponseStage(event)
             if (isInResponseStage) {
-                this.handlePausedRequestInResponseStage(event)
+                this.handlePausedRequestInResponseStage(event, cdpSession)
             } else {
-                this.handlePausedRequestInRequestStage(event)
+                this.handlePausedRequestInRequestStage(event, cdpSession)
             }
         })
     }
 
     /**
-     * Called when disconnecting from a Puppeteer.page.
+     * Called when the adapter is disconnected from
```

Comment:
```suggestion
     * Called when the adapter is disconnected from.
```
Add period for clarity (looks like an unfinished sentence)